problems/6-dude-where/problem.md

```diff
@@ -1,41 +1,59 @@
 
-A CAR file contains a DAG, a collection of nodes created from your data. There's loads of benefits to this - for example, you can retrieve sub-graphs and fetch different nodes from different peers, spreading the load and optimising speed.
+## The Challenge:
+A CAR file contains a DAG—a graph of nodes that represent your data. When you fetch data over HTTP, it’s all nice and tidy, but there’s a catch: you can’t just trust the bytes you receive. You need to verify that what you got actually matches the CID you were expecting. This means re-assembling the data into its original DAG structure and making sure everything checks out.
 
-However, when you access data from a gateway, IPFS fetches the nodes of the graph, verifies their individual hashes, re-assembles the file and sends it to you over HTTP.
-
-You receive a bunch of bytes, but the problem is you can't verify those bytes correspond to your CID, without transforming them back into a DAG 🙈.
-
-...and then, guess what, there's not just one way to build a DAG!
-
-So even if you did build a new DAG, you may not be using the same settings as whoever built it in the first place and you might come up with a different CID to the one you accessed the content with!
-
-One answer is to move the re-assembily from DAG to file into the client - the entity requesting the data. This removes the trust element as the client can now verify the hashes of the nodes in the graph.
-
-In this exercise you're going to fetch a CAR file from a "trustless" gateway (I know that sounds bad but actually it's good!) and re-assemble it into a file.
-
-Create a _new_ file for your solution e.g. `ex7.mjs`. You're going to need a couple of libraries, one for reading CARs and one for exporting files from an IPFS DAG:
-
-```sh
+## What You Need to Do:
+1. Yes, you guessed it - file number 7 - ex7.mjs, for this challenge.
+2. Install Necessary Libraries: You’ll need two libraries to get this done: @ipld/car for reading CAR files and ipfs-unixfs-exporter for exporting files from an IPFS DAG.
+```js
 npm install @ipld/car
 npm install ipfs-unixfs-exporter
 ```
+3. Fetch the CAR File: You’ll be fetching a CAR file from the web3.storage trustless gateway. The CID for the directory containing the file you want is `bafybeifdnnn35jnwel2dno7hu7zfew6gtxhkaynaszuj4i77j4m7wsv72a`, and the specific file to extract is `/cat2.txt`.
+```js
+const cid = 'bafybeifdnnn35jnwel2dno7hu7zfew6gtxhkaynaszuj4i77j4m7wsv72a'
+const path = '/cat2.txt'
 
-You can fetch a CAR from the web3.storage trustless gateway like this:
-
+const res = await fetch(`https://dag.w3s.link/ipfs/${cid}?format=car`)
+const bytes = new Uint8Array(await res.arrayBuffer())
+```
+4. Read the CAR File: Use CarReader from @ipld/car to read the CAR file. This step is crucial because you need to adapt this reader to work with the exporter from ipfs-unixfs-exporter.
```

Comment:
```suggestion
4. Read the CAR File: Use CarReader from `@ipld/car` to read the CAR file. This step is crucial because you need to adapt this reader to work with the exporter from `ipfs-unixfs-exporter`.
```